elastic/apm-server
# Don't set `processor.event` for logs

## Motivation/summary

This is not used in any queried made by the APM UI, and is an APM-specific field that we should phase out. By removing it from logs, we can potentially write logs to non-APM specific log datastreams.

See also https://github.com/elastic/kibana/issues/161718

## Checklist

~- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)~
~- [ ] Documentation has been updated~

## How to test these changes

1. Send some traces and logs, e.g. using https://github.com/elastic/apm-server/tree/main/systemtest/cmd/sendotlp
2. Check that the logs data stream does not have a `processor.event` field
3. Check that the logs still show up in the APM UI

## Related issues

https://github.com/elastic/kibana/issues/161718
